src/ansys/speos/core/simulation.py

```diff
@@ -788,6 +792,54 @@ def set_automatic_save_frequency(self, value: int = 1800) -> SimulationDirect:
         self._job.direct_mc_simulation_properties.automatic_save_frequency = value
         return self
 
+    def set_light_expert(self, value: bool = False, ray_number: int = 10e6) -> SimulationDirect:
+        """Activate/Deactivate the generation of light expert file.
+
+        Parameters
+        ----------
+        value : bool
+            Activate/Deactivate.
+            By default, ``False``, means deactivate.
+        ray_number : int
+            number of rays stored in lpf file
+            By default, ``10e6``
+
+        Returns
+        -------
+        ansys.speos.core.simulation.SimulationDirect
+            Interactive simulation
+        """
+        self._light_expert_changed = True
+        warnings.warn(
+            "Please note by Setting a value for light expert option, this will force a sensor"
+            "commit on committing the Simulation class",
+            stacklevel=2,
+        )
+        if value:
+            for item in self._project._features:
+                if isinstance(item, BaseSensor):
+                    item.lxp_path_number = ray_number
+        else:
+            for item in self._project._features:
+                if isinstance(item, BaseSensor):
+                    item.lxp_path_number = None
+        return self
+
+    def commit(self) -> SimulationDirect:
+        """Save feature: send the local data to the speos server database.
+
+        Returns
+        -------
+        ansys.speos.core.simulation.SimulationDirect
+            Simulation feature.
+        """
+        if self._light_expert_changed:
+            for item in self._project._features:
+                if isinstance(item, BaseSensor):
+                    item.commit()
```

Comment:
```suggestion
                    item.commit()
             self._light_expert_changed = False
```
in the end of the if statement, change the self._light_expert_changed = False to avoid committing more local changed.